### PR TITLE
AndesMessage

### PR DESCRIPTION
**Pull request by nrostanml**
## Motivation
Implement the AndesMessage component defined by the UX team
## Description
AndesMessage is a component defined [here](https://company-161429.frontify.com/d/kxHCRixezmfK/n-a#/componentes/message).

AndesMessage follows the guidelines of previous components, it has a public API (AndesMessage.swift) and internal views that are instantiated depending on the message variant.
AndesMessage configures each variant through AndesMessageViewConfig, which is a struct that contains all the needed variables to set up the internal views. and each internal view must implement the AndesMessageViewDelegate to pass events back to AndesMessage.

## Affected component
AndesMessage (new)
## Frontify component link
 [link](https://company-161429.frontify.com/d/kxHCRixezmfK/n-a#/componentes/message).
## Screenshots
Screen Shot 2020-01-24 at 12 23 24
Screen Shot 2020-01-24 at 12 23 37
message

## Checks
Are you sure you followed all those steps? If so, repeat after me "I solemnly swear that ...":
   - [x] I have coded an example inside the Demo App,
   - [x] I have added proper tests,
   - [x] I have modified the Changelog.md file,
   - [x] I have updated GitHub wiki,
   - [x] I have the explicit approval of one or more members of the UX Team,
   - [x] I have checked that this code is ObjC compliant.
## Change type
This is easy, let us know what this code is about:
   - [x] This code adds a new component
   - [ ] This code includes improvements to an existent component
   - [ ] This code improves or modifies ONLY the Demo App.

### CODE REVIEW

**Comment by furyGorgoryAdmin** [comment]
### Informative Menu <h2>  ### 

### Use any of these tags in a pull request description or leave a comment ### 
| Tag | Description |
|---|---|
| **[canary]**| Creates a build that can only be deployed with canary strategy. | 
| **[build]** |  Creates version or build from a branch that has been tagged as "no releasable". | 
| **[name:version-name]** |Specify the name of the build. For example: [build] [name:1.2.3-name] or [canary] [name:1.2.3-name].|
| **[major]** | To automatically increase the first position of the version. For example: [build] [major] or [canary] [major].|
| **[minor]** | To automatically increase the second position of the version. For example: [build] [minor] or [canary] [minor]. |
| **[patch]** | To automatically increase the third position of the version. For example: [build] [patch] or [canary] [patch].|  

[For more information about ci-cd](http://docs.furycloud.io/?lang=esp&section=businessassurance)
[For more information about semver](https://semver.org/#summary)

File: Example/Tests/TestHelpers.swift [hunk 1]
```diff
@@ -0,0 +1,9 @@
+//
+//  TestHelpers.swift
+//  AndesUI_Example
+//
+//  Created by Nicolas Rostan Talasimov on 1/17/20.
+//  Copyright © 2020 MercadoLibre. All rights reserved.
+//
+
+import UIKit
```
Comment (LFuryk) [inline Example/Tests/TestHelpers.swift]: Esta file?

Comment (nrostanml) [reply Example/Tests/TestHelpers.swift]: removed

File: LibraryComponents/Classes/AndesMessage/View/AndesMessageViewDelegate.swift [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+//  AndesMessageViewDelegate.swift
+//  AndesUI
+//
+//  Created by Nicolas Rostan Talasimov on 1/17/20.
+//
+
+import Foundation
+
+/// Used to route events form internal message views to AndesMessage
+internal protocol AndesMessageViewDelegate: AnyObject {
```
Comment (LFuryk) [inline LibraryComponents/Classes/AndesMessage/View/AndesMessageViewDelegate.swift]: porque extiende a Any?

File: LibraryComponents/Classes/AndesMessage/Config/AndesMessageViewConfig.swift [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    var dismissIconColor: UIColor = AndesStyleSheetManager.styleSheet.textWhiteColor
+    var primaryActionText: String?
+    var secondaryActionText: String?
+
+    //TODO:- REMOVE THIS WHEN ICONS ADDED
+    static func getDismissIcon() -> UIImage {
+        if #available(iOS 13.0, *) {
+            return UIImage(systemName: "multiply")!
+        } else {
+            return UIImage(named: "logo_global")!
```
Comment (LFuryk) [inline LibraryComponents/Classes/AndesMessage/Config/AndesMessageViewConfig.swift]: En la testapp se agrega un icono que es transparente que te va a servir mejor para chequear estos casos, se llama "clip" y entra con este [PR](https://github.com/mercadolibre/fury_andesui-ios/pull/5)

File: LibraryComponents/Classes/AndesMessage/AndesMessage.swift [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        return self
+    }
+
+    /// Should return a view depending on which message variant is selected
+    private func provideView() -> AndesMessageView {
+        return AndesMessageDefaultView(withConfig: provideConfig())
+    }
+
+    /// Should return a config depending on which message variant is selected
+    private func provideConfig() -> AndesMessageViewConfig {
```
[the remark this answers is not in the document]
Comment (nrostanml) [reply LibraryComponents/Classes/AndesMessage/AndesMessage.swift]: done

File: LibraryComponents/Classes/AndesMessage/View/AndesMessageAbstractView.swift [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    func update(withConfig config: AndesMessageViewConfig) {
+        self.config = config
+        updateView()
+    }
+
+    func setup() {
+        loadNib()
+        addSubview(messageView)
+        messageView.frame = self.bounds
+        autoresizingMask = [.flexibleWidth, .flexibleHeight]
```
[the remark this answers is not in the document]
Comment (nrostanml) [reply LibraryComponents/Classes/AndesMessage/View/AndesMessageAbstractView.swift]: es necesario porque es para la view cargada del xib, el autolayout era para "linkear" AndesMessage con esta view (jaja son 3 views en total AndesMessage, AndesMessageAbstractView y messageVIew (que es cargada por nib)) ahora esta AndesMessage <(AutoLayout)> AndesMessageAbstractView <AutoRezisingMask> messageView, lo voy a hacer todo con autolayout porque AutoRezisingMask deberia ser deprecado antes jaja

File: LibraryComponents/Classes/AndesMessage/View/AndesMessageViewDelegate.swift (hunk 2 above)
Comment (nrostanml) [reply LibraryComponents/Classes/AndesMessage/View/AndesMessageViewDelegate.swift]: para poder hacer weak el delegate

File: LibraryComponents/Classes/AndesMessage/Config/AndesMessageViewConfig.swift (hunk 3 above)
Comment (nrostanml) [reply LibraryComponents/Classes/AndesMessage/Config/AndesMessageViewConfig.swift]: como hablamos por slack, no aplica porque los iconos son fijos y tienen que estar en la lib

File: LibraryComponents/Classes/AndesMessage/AndesMessage.swift [hunk 6]
```diff
@@ -121,9 +121,6 @@ import UIKit
 
 extension AndesMessage: AndesMessageViewDelegate {
     func dismissTapped() {
-        self.view.removeFromSuperview()
```
Comment (LFuryk) [inline LibraryComponents/Classes/AndesMessage/AndesMessage.swift]: esto a donde lo movimos?

**Pull request merged**